Assets/Global/Scripts/Player/Tail.cs

```diff
@@ -1,14 +1,31 @@
 using System;
 using UnityEngine;
-
+using System.Collections.Generic;
 public class Tail : MonoBehaviour
 {
     public Player player;
 
+    private List<Collider> colliders;
+    public void Start()
+    {
+        colliders = new List<Collider>();
+    }
+    public void Update()
+    {
+        colliders = player.tailCanDoDamage? colliders : new List<Collider>();
```

Comment:
I'd prefer not to make a new List every update.

Reply:
True, It will defenitly help a bit with performance (though computers will handle it either way)

In c# (or other garbage collection languages for that matter)
you actually only want to re-assign the collection variables (lists/arrays/dictionaries) if there is a good reason for it. E.g. replacing collection A with B. Or preveting lingering references if the list is used as input for other actions. 
But in our case we just use it locally, and only add to it, or clear it.
In cases like this its better to make the collection readonly, and instead use clear (and maybe trim? not sure about that though).  (doing it that way, we dont constantly re-allocate new memory for a new list)

so you get:
`private readonly List<Collider> colliders;`
`colliders.Clear();`

Reply:
Btw, even when you use clear. Dont do it in the `update()` function
Instead, i would just do it whenever you enter the attack state.